core/instance.go

```diff
@@ -222,16 +222,16 @@ func (i *instance) terminate() error {
 	return err
 }
 
-func (i *instance) shouldBeReplacedWithSpot() bool {
+func (i *instance) shouldBeReplacedWithSpot(doNotCheckEventEnabled bool) bool {
```

Comment:
why do we need this new `doNotCheckEventEnabled` flag?

Reply:
i have explained it here:
https://github.com/mello7tre/AutoSpotting/commit/710149391d83ef4e11e7e5536563eb01da4d0676#commitcomment-49924628

As now all replacement are handled by sqsqueue and the method excuted to replace is always `swapWithGroupMember` i need this flag to let replacement work even if ASG is not enabled to process events (only cron schedule).
For spot start event i check that ASG is enabled or not before sending message to sqsqueue.